tools/protodoc/protodoc.py

```diff
@@ -194,16 +214,43 @@ def FormatExtension(extension):
     anchor = FormatAnchor('extension_' + extension)
     status = EXTENSION_STATUS_VALUES.get(extension_metadata['status'], '')
     security_posture = EXTENSION_SECURITY_POSTURES[extension_metadata['security_posture']]
+    categories = FormatExtensionList(extension_metadata["categories"], "extension_category")
     return EXTENSION_TEMPLATE.substitute(anchor=anchor,
                                          extension=extension,
                                          status=status,
-                                         security_posture=security_posture)
+                                         security_posture=security_posture,
+                                         categories=categories)
   except KeyError as e:
     sys.stderr.write(
         '\n\nDid you forget to add an entry to source/extensions/extensions_build_config.bzl?\n\n')
     exit(1)  # Raising the error buries the above message in tracebacks.
 
 
+def FormatExtensionList(items, prefix="extension", indent=2):
+  indent = " " * indent
+  formatted_list = "\n".join(f"{indent}- :ref:`{ext} <{prefix}_{ext}>`" for ext in items)
+  return f"{formatted_list}\n{indent}\n"
+
+
+def FormatExtensionCategory(extension_category):
+  """Format extension metadata as RST.
+
+  Args:
+    extension_category: the name of the extension_category, e.g. com.acme.
+
+  Returns:
+    RST formatted extension category description.
+  """
+  try:
+    extensions = EXTENSION_CATEGORIES[extension_category]
+  except KeyError as e:
+    sys.stderr.write(f"\n\nUnable to find extension category:  {extension_category}\n\n")
+    exit(1)  # Raising the error buries the above message in tracebacks.
```

Comment:
Suggest turning the KeyError into a ProtodocError and then throwing.

Reply:
the problem is that we need the `exit` - i can make protodocerror inherit `SysExit` tho - should i ?

Reply:
hmm - maybe it *will* exit anyway - ill test...

Reply:
cool - raising `ProtodocError` actually gives better err/tb - updated